jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLClasspathHelperImpl.java

```diff
@@ -239,6 +246,7 @@ public synchronized void addClasspathEntry(URL cpe) {
 				invalidURLExists = true;
 			}
 			++urlCount;
+			notifyClasspathChange3(id, loader, urls, urlCount, true);
```

Comment:
This can be optimized to pass in the index of the new URL, as only the new URL needs to have a timestamp check.

Reply:
Already modified as discussed. Need to compile a build to see whether it still works good.